src/streamdiffusion/wrapper.py

```diff
@@ -1773,11 +1773,17 @@ def update_control_image(self, index: int, image: Union[str, Image.Image, torch.
             logger.debug("update_control_image: Skipping ControlNet update in skip diffusion mode")
 
 
-    def update_style_image(self, image: Union[str, Image.Image, torch.Tensor], is_stream: bool = False) -> None:
+    def update_style_image(self, image: Union[str, Image.Image, torch.Tensor], is_stream: bool = False, style_key = "ipadapter_main") -> None:
         """Update IPAdapter style image"""
         if not self.use_ipadapter:
             raise RuntimeError("update_style_image: IPAdapter support not enabled. Set use_ipadapter=True in constructor.")
-        self.stream.update_style_image(image, is_stream=is_stream)
+        
+        if not self.skip_diffusion:
```

Comment:
Is this `skip_diffusion` updatable through `update_stream_params`? If so, probably worth not skipping this update if it's disabled as it could be enabled right after. If the caller doesn't want the style image to be updated they should not call this function.